### PR TITLE
Init readme

### LINKED ISSUE #1: Update README.md
Currently, the Readme contains the boilerplate contents of CRA. That needs to be replaced with appropriate content for the project.

### PR DESCRIPTION
**Pull request by ritikrishu**
Fixes #1 #2 

#### Checklist

- [ ] I have read the Contribution & Best practices Guide and my PR follows them.(not applicable)
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added/updated necessary tests/documentation (if applicable)

#### Changes proposed in this pull request:
this is initial scaffolding for docs.
PR separates code into _client_ and _server_ folders for respective modules.
EDIT:
This PR is highly open to suggestions and improvements

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2444 +1,93 @@
...
-
-![Jest watch mode](http://facebook.github.io/jest/img/blog/15-watch.gif)
-
-### Version Control Integration
-
-By default, when you run `npm test`, Jest will only run the tests related to files changed since the last commit. This is an optimization designed to make your tests run fast regardless of how many tests you have. However it assumes that you don’t often commit the code that doesn’t pass the tests.
-
-Jest will always explicitly mention that it only ran tests related to the files changed since the last commit. You can also press `a` in the watch mode to force Jest to run all tests.
-
-Jest will always run all tests on a [continuous integration](#continuous-integration) server or if the project is not inside a Git or Mercurial
```
Comment (ritikrishu) [inline README.md]: @divyamrastogi we'll need your input here.

Comment (ritikrishu) [inline README.md]: @divyamrastogi we need your input [here about server module](https://github.com/ritikrishu/showcase-app/blob/ini-readme/README.md#user-content-project-set-up).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+     git checkout -b my-fix-branch master
+     ```
+4. Create your patch, **including appropriate test cases**.
+5. Run the full project test suite, as described in the [readme](https://github.com/UdacityMobileWebScholarship/showcase-app/tree/master#user-content-project-set-up),
+  and ensure that all tests pass.
+6. Commit your changes using a descriptive commit message. ex-
+```shell
+    git commit -m "doc: update readme"
+    git commit -m "ref: refactor API service"
+    git commit -m "feat: add authentication service"
```
Comment (palnabarun) [inline CONTRIBUTING.md]: I am actually against commit messages in this tense. The messages should look like `Updates readme`, `Refactors API Service`, `Adds authentication service`. I hope you get my point.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+* **build**: Changes that affect the build system or external dependencies (example scopes: webpack, npm, deploy script)
+* **ci**: Changes to our CI configuration files and scripts (example scopes: Travis, Circle)
+* **docs**: Documentation only changes
+* **feat**: A new feature
+* **fix**: A bug fix
+* **perf**: A code change that improves performance
+* **ref**: A code change that neither fixes a bug nor adds a feature
+* **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+* **test**: Adding missing tests or correcting existing tests
+* **UI**: UI related additions
```
Comment (palnabarun) [inline CONTRIBUTING.md]: I think we should not add labels to the commit messages as the history sometimes looks weird while looking at very early commits. This categorization should ideally be done at the PR level.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+Before you submit an issue, please search the issue tracker, maybe an issue for your identified problem already exists and you can join the discussion there only.
+For bugs, please try to provide as much information possible to reproduce it as possible, that will really help us a lot!
+
+### <a name="submit-pr"></a> Submitting a Pull Request (PR)
+1. Make sure there is an issue open for the PR and it is assigned to you before starting any work. That way it'll be helpful to eliminate duplicate efforts.
+2. Fork the repo.
+3. Make your changes in a new git branch:
+
+     ```shell
+     git checkout -b my-fix-branch master
```
Comment (palnabarun) [inline CONTRIBUTING.md]: We should add a note here, asking the collaborator to update his master once again before checking out the feature branch. Many times people forget to do this. I am also a culprit sometimes. :)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # Contributing to The Show Case App
 
-We would love for you to contribute to the project and help make it even better! You can contribute either through code or helping us out on different issues, suggesting features etc. As a contributor here are the guidelines we would like you to follow:
+We would love for you to contribute to the project and help make it even better! You can contribute either through code or helping us out on different issues, suggesting features etc. As a contributor here is the guidelines we would like you to follow:
```
Comment (palnabarun) [inline CONTRIBUTING.md]: The previous version was the correct one. Don't judge yourself that much. :)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -10,19 +10,19 @@ We would love for you to contribute to the project and help make it even better!
  - [Submission Guidelines](#submit)
 
 ## <a name="coc"></a> Code of Conduct
-Help us keep the project open and inclusive. Please read and follow our [Code of Conduct][https://github.com/UdacityMobileWebScholarship/showcase-app/blob/master/CODE_OF_CONDUCT.md].
+Help us keep the project open and inclusive. Please read and follow our (Code of Conduct)[https://github.com/UdacityMobileWebScholarship/showcase-app/blob/master/CODE_OF_CONDUCT.md].
```
Comment (palnabarun) [inline CONTRIBUTING.md]: I think this the correct one `[  ] ( )` [google](google.com)

File: CONTRIBUTING.md (hunk 3 above)
Comment (ritikrishu) [reply CONTRIBUTING.md]: This is optional as mentioned.

File: CONTRIBUTING.md (hunk 4 above)
Comment (ritikrishu) [reply CONTRIBUTING.md]: Good point. Totally agree!

File: CONTRIBUTING.md (hunk 6 above)
Comment (ritikrishu) [reply CONTRIBUTING.md]: typo😃

File: CONTRIBUTING.md (hunk 2 above)
Comment (palnabarun) [reply CONTRIBUTING.md]: Actually, new collaborators tend to follow what is written in the Contribution Guidelines. We should also add links to some commit best practices.

File: CONTRIBUTING.md (hunk 5 above)
Comment (ritikrishu) [reply CONTRIBUTING.md]: This commit is done by Grammarly not me! 😅

File: CONTRIBUTING.md (hunk 2 above)
Comment (ritikrishu) [reply CONTRIBUTING.md]: That would be great! Can you add some or suggest so I can add!

Comment (PrabhanshuAttri) [reply CONTRIBUTING.md]: Git branch naming practices can also be included to segregate Work in Progress(wip), bug, feature etc. [Source](https://stackoverflow.com/questions/273695/git-branch-naming-best-practices?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa)

Comment (palnabarun) [reply CONTRIBUTING.md]: I mostly use `issuenumber-someverbosetext` as branch names. It helps me remember which issue I am addressing. Like `1852-check-input-type-select-atoms`.

File: CONTRIBUTING.md (hunk 5 above)
Comment (palnabarun) [reply CONTRIBUTING.md]: Grammarly makes commits? Whoa. I need that integration. :)

File: CONTRIBUTING.md (hunk 2 above)
Comment (ritikrishu) [reply CONTRIBUTING.md]: I like the idea of issue number prefix!

File: CONTRIBUTING.md (hunk 5 above)
Comment (ritikrishu) [reply CONTRIBUTING.md]: That was a figure of speech not literal statement 😛

File: CONTRIBUTING.md (hunk 2 above)
Comment (ritikrishu) [reply CONTRIBUTING.md]: @PrabhanshuAttri `wip:`, `feat:` can be included in commit msg as well, it makes more sense there.
adding labels to issues/PR can help us filter category wise!

**Comment by ritikrishu** [comment]
Commits can be squashed during merge. IMO, it's better to go through whole build process of PR commit by commit. @ParthS007

File: CONTRIBUTING.md (hunk 2 above)
Comment (palnabarun) [reply CONTRIBUTING.md]: @ritikrishu Exactly. The commit message should ideally look like this.

```
A small 50 character line

Long description describing the change
with keywords for filtering and all
```

For that we need to enforce using `git commit` without `-m` switch. The collaborator can use `git commit -m` when the information is small and can be written in less than 50 characters.

File: CONTRIBUTING.md (hunk 5 above)
Comment (palnabarun) [reply CONTRIBUTING.md]: I was joking, sir. :P

File: CONTRIBUTING.md (hunk 2 above)
Comment (ritikrishu) [reply CONTRIBUTING.md]: this is a minimal example for those who are not well well versed with git. We'd want them to learn along the way. Contributors can always look into other PRs who've been merged for reference. We can even add one or two links in there when we have such code contribution PRs.

**Comment by ParthS007** [comment]
@divyamrastogi @palnabarun  Please review this. We can further make the changes if there are any :+1:

**Comment by palnabarun** [comment]
Seems good to me at the moment. Let's add anything remaining later.

**Comment by realslimshanky** [comment]
Thanks for your contribution @ritikrishu :tada:

**Pull request merged**